Client/Frontend/Reader/ReaderModeCache.swift

```diff
@@ -18,69 +18,77 @@ class ReaderModeCache {
+            let contentFilePath = (cacheDirectoryPath as NSString).stringByAppendingPathComponent("content.json")
+            let string: NSString = readabilityResult.encode()
+            try string.writeToFile(contentFilePath, atomically: true, encoding: NSUTF8StringEncoding)
+            return
+        } catch let error1 as NSError {
+            throw error1
         }
-        return false
     }
 
-    func get(url: NSURL, error: NSErrorPointer) -> ReadabilityResult? {
-        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
-            let contentFilePath = cacheDirectoryPath.stringByAppendingPathComponent("content.json")
-            if NSFileManager.defaultManager().fileExistsAtPath(contentFilePath) {
-                if let string = NSString(contentsOfFile: contentFilePath, encoding: NSUTF8StringEncoding, error: error) {
-                    return ReadabilityResult(string: string as String)
+    func get(url: NSURL) throws -> ReadabilityResult {
+        var error: NSError! = NSError(domain: "Migrator", code: 0, userInfo: nil)
+        let cacheDirectoryURL = url.URLByAppendingPathComponent("content.json")
+        guard let contentFilePath = cacheDirectoryForURL(cacheDirectoryURL) else {
+            throw error
+        }
+
+        if NSFileManager.defaultManager().fileExistsAtPath(contentFilePath) {
+            do {
+                let string = try NSString(contentsOfFile: contentFilePath, encoding: NSUTF8StringEncoding)
+                if let value = ReadabilityResult(string: string as String) {
+                    return value
                 }
+            } catch let error1 as NSError {
+                error = error1
             }
         }
-        return nil
+        throw error
     }
 
     func delete(url: NSURL, error: NSErrorPointer) {
-        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
-            if NSFileManager.defaultManager().fileExistsAtPath(cacheDirectoryPath) {
-                NSFileManager.defaultManager().removeItemAtPath(cacheDirectoryPath, error: error)
+        guard let cacheDirectoryPath = cacheDirectoryForURL(url) else { return }
+        if NSFileManager.defaultManager().fileExistsAtPath(cacheDirectoryPath) {
+            do {
+                try NSFileManager.defaultManager().removeItemAtPath(cacheDirectoryPath)
+            } catch let error1 as NSError {
+                error.memory = error1
             }
         }
     }
 
-    func contains(url: NSURL, error: NSErrorPointer) -> Bool {
-        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
-            let contentFilePath = cacheDirectoryPath.stringByAppendingPathComponent("content.json")
-            return NSFileManager.defaultManager().fileExistsAtPath(contentFilePath)
+    func contains(url: NSURL) throws {
+        let error: NSError! = NSError(domain: "Migrator", code: 0, userInfo: nil)
```

Comment:
This is the kind of shit introduced by the migration tool. I hate
autogenerated code.

On 27 August 2015 at 18:55, Stephan Leroux <notifications@github.com> wrote:

> In Client/Frontend/Reader/ReaderModeCache.swift
> <https://github.com/mozilla/firefox-ios/pull/966#discussion_r38125258>:
>
> >              }
> >          }
> >      }
> >
> > -    func contains(url: NSURL, error: NSErrorPointer) -> Bool {
> > -        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
> > -            let contentFilePath = cacheDirectoryPath.stringByAppendingPathComponent("content.json")
> > -            return NSFileManager.defaultManager().fileExistsAtPath(contentFilePath)
> > +    func contains(url: NSURL) throws {
> > +        let error: NSError! = NSError(domain: "Migrator", code: 0, userInfo: nil)
>
> Also just noticed, when using error domains, they are usually in a reverse
> url format. So likee this one could be
> com.mozilla.readermodecache.migrator. Might also make sense to create a
> enum with various error codes so it's easy to track what the errors mean
> when we throw them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/firefox-ios/pull/966/files#r38125258>.
>